src/Stratis.Bitcoin.Features.Dns.Tests/GivenADnsFeature.cs

```diff
@@ -261,10 +261,12 @@ public void WhenDnsServerFailsToStart_ThenDnsFeatureRetries()
                 .Setup(f => f.Log(It.IsAny<LogLevel>(), It.IsAny<EventId>(), It.IsAny<It.IsAnyType>(), It.IsAny<Exception>(), (Func<It.IsAnyType, Exception, string>)It.IsAny<object>()))
                 .Callback(new InvocationAction(invocation =>
                 {
-                    if (!serverError && (LogLevel)invocation.Arguments[0] == LogLevel.Error)
+                    if (!serverError && (LogLevel)invocation.Arguments[0] == LogLevel.Error && invocation.Arguments[4] is Func<object[], Exception, string> func)
```

Comment:
Are all these changes part of the logging PR?

Reply:
Yes. These tests, for some reason, expect to find the log message in the string formatting parameters. This fixes that assumption by looking at the resultant formatted string instead.

Reply:
Reverted.